internal/services/eventhub/migration/consumer_group.go

```diff
@@ -0,0 +1,80 @@
+package migration
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/azure"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/eventhub/sdk/2017-04-01/consumergroups"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+var _ pluginsdk.StateUpgrade = ConsumerGroupsV0ToV1{}
+
+type ConsumerGroupsV0ToV1 struct{}
+
+func (ConsumerGroupsV0ToV1) UpgradeFunc() pluginsdk.StateUpgraderFunc {
+	return func(ctx context.Context, rawState map[string]interface{}, meta interface{}) (map[string]interface{}, error) {
+		// old:
+		// 	/subscriptions/12345678-1234-5678-1234-123456789012/resourceGroups/group1/providers/Microsoft.EventHub/namespaces/namespace1/eventhubs/eventhub1/consumergroups/consumergroup1
+		// new:
+		// 	/subscriptions/12345678-1234-5678-1234-123456789012/resourceGroups/group1/providers/Microsoft.EventHub/namespaces/namespace1/eventhubs/eventhub1/consumerGroups/consumergroup1
+		oldId, err := azure.ParseAzureResourceID(rawState["id"].(string))
+		if err != nil {
+			return rawState, err
+		}
+
+		consumerGroupSegment := ""
+		for key, value := range oldId.Path {
+			if strings.EqualFold(key, "consumergroups") {
+				consumerGroupSegment = value
+				break
+			}
+		}
+
+		if consumerGroupSegment == "" {
+			return rawState, fmt.Errorf("couldn't find the `consumerGroups` segment in the old resource id %q", oldId)
+		}
+
+		newId := consumergroups.NewConsumerGroupID(oldId.SubscriptionID, oldId.ResourceGroup, oldId.Path["namespaces"], oldId.Path["eventhubs"], consumerGroupSegment)
```

Comment:
the parser has an Insensitively function, so we can make this:

```suggestion
		oldId := rawState["id"].(string)
		parsed, err := ParseConsumerGroupIDInsensitively(oldId)
		if err != nil {
			return fmt.Errorf("parsing existing Consumer Group ID %q: %+v", oldId, err)
		}
		
		newId := parsed.ID()
```

Reply:
ohh nifty!